`make ut` fails: ERROR: unsatisfiable constraints: py2-pip (missing)

Make target for unit tests executions fails, because 'calico/test' image can't be built:

> + make ut
> docker build -f Dockerfile.calico_test -t calico/test:latest .
> Sending build context to Docker daemon 557.1 kB
> Sending build context to Docker daemon 618.5 kB
> 
> Step 1 : FROM docker
>  ---> 9b2086b6e30d
> Step 2 : MAINTAINER Tom Denham <tom@projectcalico.org>
>  ---> Using cache
>  ---> 0f09ac930f23
> Step 3 : RUN apk add --update python python-dev py2-pip py-setuptools openssl-dev libffi-dev         git musl-dev gcc tshark netcat-openbsd         iptables ip6tables iproute2 iputils ipset curl &&         curl -o glibc.apk -L "https://github.com/andyshinn/alpine-pkg-glibc/releases/download/2.23-r1/glibc-2.23-r1.apk" &&         apk add --allow-untrusted glibc.apk &&         curl -o glibc-bin.apk -L "https://github.com/andyshinn/alpine-pkg-glibc/releases/download/2.23-r1/glibc-bin-2.23-r1.apk" &&         apk add --allow-untrusted glibc-bin.apk &&         /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc/usr/lib &&         echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf &&         rm -f glibc.apk glibc-bin.apk &&         rm -rf /var/cache/apk/*
>  ---> Running in 387115cd2ce1
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
> fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
> ERROR: unsatisfiable constraints:
>   py2-pip (missing):
>     required by: world[py2-pip]
> The command '/bin/sh -c apk add --update python python-dev py2-pip py-setuptools openssl-dev libffi-dev         git musl-dev gcc tshark netcat-openbsd         iptables ip6tables iproute2 iputils ipset curl &&         curl -o glibc.apk -L "https://github.com/andyshinn/alpine-pkg-glibc/releases/download/2.23-r1/glibc-2.23-r1.apk" &&         apk add --allow-untrusted glibc.apk &&         curl -o glibc-bin.apk -L "https://github.com/andyshinn/alpine-pkg-glibc/releases/download/2.23-r1/glibc-bin-2.23-r1.apk" &&         apk add --allow-untrusted glibc-bin.apk &&         /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc/usr/lib &&         echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf &&         rm -f glibc.apk glibc-bin.apk &&         rm -rf /var/cache/apk/*' returned a non-zero code: 1
> Makefile:45: recipe for target 'calico_test.created' failed
> make: *** [calico_test.created] Error 1

The failure is caused by an old base image (docker) which is based on alpine linux v3.4 that lacks 'py2-pip' package. Current 'latest' points to 3.5 version which has 'py2-pip', so `docker pull docker` fixes the issue locally. I think base image in Dockerfile.calico_test should be pinned to some version in order to prevent such failures in the future.

--- a/Dockerfile.calico_test
+++ b/Dockerfile.calico_test
@@ -33,7 +33,7 @@
 # - eliminate most isolation, (--uts=host --pid=host --net=host --privileged)
 # - volume mount your ST source code
 # - run 'nosetests'
-FROM docker
+FROM docker:1.13.0
 MAINTAINER Tom Denham <tom@projectcalico.org>
 
 # Running STs in this containers require that it has all dependencies installed

